include/dlaf/eigensolver/reduction_to_band/impl.h

```diff
@@ -680,33 +712,96 @@ void her2kUpdateTrailingMatrix(const LocalTileSize& at_start, MatrixT<T>& a,
 
       // The first column of the trailing matrix (except for the very first global tile) has to be
       // updated first, in order to unlock the next iteration as soon as possible.
-      const auto& ex =
-          (j == at_start.cols()) ? getHpExecutor<Backend::MC>() : getNpExecutor<Backend::MC>();
+      const auto& priority = (j == at_start.cols()) ? thread_priority::high : thread_priority::normal;
 
       if (is_diagonal_tile) {
-        her2kDiag(ex, v.read(ij_local), x.read(ij_local), a(ij_local));
+        her2kDiag<B>(priority, v.read_sender(ij_local), x.read_sender(ij_local),
+                     a.readwrite_sender(ij_local));
       }
       else {
         // A -= X . V*
-        her2kOffDiag(ex, x.read(ij_local), vt.read(ij_local), a(ij_local));
+        her2kOffDiag<B>(priority, x.read_sender(ij_local), vt.read_sender(ij_local),
+                        a.readwrite_sender(ij_local));
 
         // A -= V . X*
-        her2kOffDiag(ex, v.read(ij_local), xt.read(ij_local), a(ij_local));
+        her2kOffDiag<B>(priority, v.read_sender(ij_local), xt.read_sender(ij_local),
+                        a.readwrite_sender(ij_local));
       }
     }
   }
 }
 }
 }
 
+template <Backend B, Device D, class T>
+struct Helper;
```

Comment:
I think a better name should be found.
In general, I'm fine with a generic name if it is a struct with just static members that is never initialized.
(the member function name is always visible when used)
When a instance of the struct is needed, the line of code with construction should be understandable and not a simple `Helper<...> helper(...)`;